core/src/main/java/org/pentaho/di/connections/vfs/provider/ConnectionFileSystem.java

```diff
@@ -70,7 +70,8 @@ public static String getUrl( AbstractFileName abstractFileName, ConnectionDetail
         domain = "/" + domain;
       }
       url = vfsConnectionDetails.getType() + ":/" + domain + abstractFileName.getPath();
-      if ( url.matches( DOMAIN_ROOT ) ) {
+      //TODO Looks like a bug. For now excluding this for 1st level roots. For future, needs to be re-analyzed.
```

Comment:
Please change comment to refer to `connections with hasBuckets` instead of `1st level roots`.

Reply:
fixed